src/hooks/useRequestParams.ts

```diff
@@ -1,93 +1,122 @@
-    } catch (e) {
-      return { request: undefined, error: e as InvalidURLParamsError };
-    }
-  }, [keys, searchParams]);
-  return request;
+// converts query params by transaction hash to parameters needed for client, or undefined
+export function getRequestInputByTransaction(
+  params: unknown
+): RequestInputByTransaction | undefined {
+  try {
+    const parsed = ss.create(params, RequestInputByTransactionSs);
+    parsed.oracleType = capitalizeFirstLetter(parsed.oracleType || defaultOracleType);
+    assert(
+      isSupportedOracleType(parsed.oracleType),
+      "Invalid oracle type in search params: " + parsed.oracleType
+    );
+    const oracleType: OracleType = parsed.oracleType;
+    return {
+      ...parsed,
+      oracleType,
+      chainId: Number(parsed.chainId),
+      eventIndex: Number(parsed.eventIndex),
+    };
+  } catch (err) {
+    //since we cant easily handle errors in hooks, just return undefined for any reason we cant parse
+  }
 }
 
-export function isByTransactionRequest(
-  request?: RequestParams<boolean>
-): request is ActiveRequestParamsByTransaction {
-  if (!request) {
-    return false;
+// converts query params by request details to parameters needed for client, or undefined
+export function getRequestInput(params: unknown): RequestInput | undefined {
+  try {
+    const parsed = ss.create(params, RequestInputSs);
+    parsed.oracleType = capitalizeFirstLetter(parsed.oracleType || defaultOracleType);
+    assert(
+      isSupportedOracleType(parsed.oracleType),
+      "Invalid oracle type in search params: " + parsed.oracleType
+    );
+    const oracleType: OracleType = parsed.oracleType;
+    return {
+      ...parsed,
+      oracleType,
+      chainId: Number(parsed.chainId),
+      timestamp: Number(parsed.timestamp),
+    };
+  } catch (err) {
+    //since we cant easily handle errors in hooks, just return undefined for any reason we cant parse
   }
-  const { chainId, transactionHash } =
-    request as ActiveRequestParamsByTransaction;
-  return chainId !== undefined && transactionHash !== undefined;
 }
 
-export class InvalidURLParamsError extends Error {
-  constructor(message: string) {
-    super("Invalid URL params");
-    this.message = message;
+// this parses the required parameters, (oracleType, chainId) from any query
+export function getRequestInputRequired(
+  params: unknown
+): RequestInputRequired | undefined {
+  try {
+    const parsed = ss.create(params, RequestInputRequiredSs);
+    parsed.oracleType = capitalizeFirstLetter(parsed.oracleType || defaultOracleType);
+    assert(
+      isSupportedOracleType(parsed.oracleType),
+      "Invalid oracle type in search params: " + parsed.oracleType
+    );
+    const oracleType: OracleType = parsed.oracleType;
+    return {
+      ...parsed,
+      oracleType,
+      chainId: Number(parsed.chainId),
+    };
+  } catch (err) {
+    //since we cant easily handle errors in hooks, just return undefined for any reason we cant parse
```

Comment:
Same here.

Reply:
i simplified thesea and removed the try catch